test/util/test/config_generator.go

```diff
@@ -105,59 +105,71 @@ func GenerateConfigurations() (config.Config, LiveConfigurations) {
 	for _, serverTLSTypeValue := range ServerTLSTypeValues() {
 		for _, listenerTypeValue := range ListenerTypeValues() {
 			for _, sslModeTypeValue := range SSlModeTypeValues() {
-				for _, sslRootCertTypeValue := range SSLRootCertTypeValues() {
-					connectionPort := ConnectionPort{
-						// TODO: perhaps resolve this duplication of listener type
-						ListenerType: listenerTypeValue,
-						Port:         portNumber,
-					}
+				for _, sslPublicCertValue := range SSLPublicCertTypeValues() {
+					for _, sslPrivateKeyValue := range SSLPrivateKeyTypeValues() {
+						for _, sslRootCertTypeValue := range SSLRootCertTypeValues() {
+							connectionPort := ConnectionPort{
+								// TODO: perhaps resolve this duplication of listener type
+								ListenerType: listenerTypeValue,
+								Port:         portNumber,
+							}
 
-					listener := config.Listener{
-						Name: "listener_" + connectionPort.ToPortString(),
-						// TODO: grab value from envvar for flexibility
-						Protocol: TestDBConfig.DB_PROTOCOL,
-						Debug: true,
-					}
-					handler := config.Handler{
-						Name: "handler_" + connectionPort.ToPortString(),
-						Debug: true,
-						ListenerName: "listener_" + connectionPort.ToPortString(),
-						Credentials: sharedCredentials(),
-					}
-					liveConfiguration := LiveConfiguration{
-						AbstractConfiguration: AbstractConfiguration{
-							ListenerType:    listenerTypeValue,
-							ServerTLSType:   serverTLSTypeValue,
-							SSLModeType:     sslModeTypeValue,
-							SSLRootCertType: sslRootCertTypeValue,
-						},
-						ConnectionPort: connectionPort,
-					}
+							listener := config.Listener{
+								Name: "listener_" + connectionPort.ToPortString(),
+								// TODO: grab value from envvar for flexibility
+								Protocol: TestDBConfig.DB_PROTOCOL,
+								Debug:    true,
+							}
+							handler := config.Handler{
+								Name:         "handler_" + connectionPort.ToPortString(),
+								Debug:        true,
+								ListenerName: "listener_" + connectionPort.ToPortString(),
+								Credentials:  sharedCredentials(),
+							}
+							liveConfiguration := LiveConfiguration{
+								AbstractConfiguration: AbstractConfiguration{
+									ListenerType:    listenerTypeValue,
+									ServerTLSType:   serverTLSTypeValue,
+									SSLModeType:     sslModeTypeValue,
+									SSLRootCertType: sslRootCertTypeValue,
+									SSLPrivateKeyType: sslPrivateKeyValue,
+									SSLPublicCertType: sslPublicCertValue,
+								},
+								ConnectionPort: connectionPort,
+							}
 
-					// sslRootCertTypeValue
-					handler.Credentials = append(handler.Credentials, sslRootCertTypeValue.toConfigVariable())
+							// sslRootCertTypeValue
+							handler.Credentials = append(handler.Credentials, sslRootCertTypeValue.toConfigVariable())
```

Comment:
`append` takes multiple arguments -- any reason not to combine all these append statements into one?

Reply:
that's true. changed